### PR TITLE
adding Topics Endpoints in project

### LINKED ISSUE #28: API: Endpoint for retrieving the list of topics
Create an endpoint for returning a list of topics.

> Note: Since there is no database yet, feel free to hardcode the list of topics just for testing purposes. Another ticket will be added to connect this endpoint to the database table.

image

### PR DESCRIPTION
**Pull request by MaherHussain**
# Description

This PR is about Endpoint for retrieving topics list.

Fixes #28 

# How to test?

i have tested it in swagger by running localhost:3000/api/documentation

# Checklist

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] This PR is ready to be merged and not breaking any other functionality

### CODE REVIEW

File: src/server/api/routes/topics.router.js [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+router.get('/', (req, res, next) => {
+  topicsController
+    .getTopics()
+    .then((result) => res.json(result))
+    .catch(next);
+});
+ 
+   /**
+    * @swagger
+    * /modules/{ID}:
```
Comment (yukafjaponesa) [inline src/server/api/routes/topics.router.js:20]: looks nice! you can change modules to topics.
fx) /topics/{ID}:
     get: summery: Get topic by Id

Comment (MaherHussain) [reply src/server/api/routes/topics.router.js]: hi Yuka . changes i done .

**Review by yukafjaponesa — COMMENTED** [review]
Hi Maher, I have the same task with different topic and Toni mentioned me we should only do what we have to do. In this case we should only do with Get and Get/ID method. You can remove post, edit, delete and other methods.

File: src/server/api/controllers/topics.controller.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
+/* TODO: This is an example controller to illustrate a server side controller.
```
Comment (tpetrina) [inline src/server/api/controllers/topics.controller.js:1]: Remove comment. Always check your PR before submitting.

File: src/server/api/routes/topics.router.js [hunk 3]
```diff
@@ -0,0 +1,151 @@
+/* TODO: This is just an example file to illustrate API routing and
```
Comment (tpetrina) [inline src/server/api/routes/topics.router.js:1]: Remove comment, read PR before submitting.

**Review by tpetrina — COMMENTED** [review]
You added way more than what was needed in the linked issue. In the future, you should read carefully what is requested from you.

**Review by yukafjaponesa — DISMISSED** [review]
super! changes look great!

**Review by tpetrina — APPROVED** [review]

**Review by nuha90 — APPROVED** [review]

**Pull request merged**